### ISSUE TITLE
Fast CGI Crashes on Hompage when Sonarr and/or Radarr is added

### ISSUE THREAD

**Issue by seanvree** [causefx/Organizr#279]
Windows 10E
Running Organizr behind IIS Reverse Proxy
PHP via Fast CGI:  PHP v 7.1 x64

Problem:

Organizr homepage will Crash with the below error when Sonarr and/or Radarr is added.  All other pages work fine (PLEX/Sabnzbd/Headphones)

however, their NORMAL tab does work, it only crashes when they are added to the homepage

This page ALSO works:  https://seanvree.com/organizr/phpinfo.php
and this:  https://seanvree.com/organizr/check.php

homepage Settings:

http://192.168.1.20:7878/radarr (I have tried every combo known w and w/o reverse proxy)
API Key:  OMIT

When problem Started:

After upgrading to windows 10 CU (FML) then rolling back. 
As a result of the Roll-back I updated SEVERAL applications, to include PHP....so not sure if the new version of PHP made this happen.  IT WAS working before on the same box.

Error:

```
HTTP Error 500.0 - Internal Server Error
C:\PHP\7.1\php-cgi.exe - The FastCGI process exited unexpectedly


Module	   FastCgiModule
Notification	   ExecuteRequestHandler
Handler	   PHP via FastCGI
Error Code	   0x000000ff
Requested URL	   https://seanvree.com:443/organizr/homepage.php
Physical Path	   C:\inetpub\wwwroot\Organizr\homepage.php
Logon Method	   Anonymous
Logon User	   Anonymous
Request Tracing Directory	   C:\inetpub\logs\failedreqlogfiles

```

**Comment by SgtBatten** [comment]
Try just 

  192.168.1.20:7878

**Comment by seanvree** [comment]
@SgtBatten - 

I have.  I've tried every combo known.

**Comment by elmerfdz** [comment]
Do you have the x86 version of Microsoft Visual C++ Redistributable installed as well?

**Comment by seanvree** [comment]
@elmerfdz - I do. However, I just looked and it seems I have two installs with very small ver differences. 

Are you saying I SHOULD have it installed, or Not?

**Comment by elmerfdz** [comment]
I'm saying you should

![image](https://cloud.githubusercontent.com/assets/22656503/25093561/b8a454ee-238a-11e7-933d-92feca4c5173.png)

**Comment by seanvree** [comment]
@elmerfdz 

I do have it installed.  No help.

**Comment by LastStarFighter** [comment]
Whats could be happening is that FastCGI can't spawn children processes to handle the multiple requests on the homepage causing PHP to crash. In Linux FastCGI can do this natively but in Windows you need PHP v7.1 'and' have to set the environment variables so it can handle the simultaneous requests.  

I was seeing random PHP crashes running Organizr's homepage using Caddy Server instead on IIS though so it might not be the same cause. 

I added these 2 settings to my own setup and PHP hasn't crashed since.

PHP_FCGI_CHILDREN=3               
PHP_FCGI_MAX_REQUESTS=128

Basically these variables will allow FastCGI to spawn 'x' amount total processes to handle any PHP requests and after 'x' amount of requests it'll kill those processes and start new ones. You can set these variables to whatever works best for your situation.

**Comment by seanvree** [comment]
@LastStarFighter  - are you talking add these to the php.ini file?

**Comment by LastStarFighter** [comment]
I added them to my System Environment Variables but since your using IIS you actually have a FastCGI settings in your IIS Manager.. if i remember correctly. From there you could add these two environment variables to your FastCGI settings.

**Comment by seanvree** [comment]
@LastStarFighter  - I found it, yeah, in IIS Fast CGI settings.  I had my max_Requests set to 10000.  Not really sure what this mean,,,,,but I set it to 128 like you said. I'll do some testing.  I also didn't have a value for FCGI_Children, so I added that as well.

**Comment by LastStarFighter** [comment]
If the children variable worked you should see extra CGI/FastCGI processes
in your task manager (4 total if you set it to 3). 1 Parent CGI/FastCGI
process and 3 children processes that the parent controls.

If your running a public website that serve's a lot of people then 128 will
be way to short since you'll be killing/re-spawning FastCGI way to often
and wasting precious resources but not an issue for a private reverse proxy
server.

**Comment by seanvree** [comment]
Wurd, really appriciate it.  Just my private server.  

THX.

**Comment by seanvree** [comment]
Not sure what I fixed, but it's working now.
